### PR TITLE
DOC: Improve docs for `jax.numpy`: `float_power` and `nextafter`

### PR DESCRIPTION
**Pull request by rajasekharporeddy**
Part of #21461

Rendered output:
- https://jax--23536.org.readthedocs.build/en/23536/_autosummary/jax.numpy.float_power.html
- https://jax--23536.org.readthedocs.build/en/23536/_autosummary/jax.numpy.nextafter.html

### CODE REVIEW

File: jax/_src/numpy/ufuncs.py [hunk 1]
```diff
@@ -872,14 +872,76 @@ def maximum(x: ArrayLike, y: ArrayLike, /) -> Array:
...
+    >>> jnp.float_power(-3, 1.7)
+    Array(nan, dtype=float32, weak_type=True)
+  """
   return lax.pow(*promote_args_inexact("float_power", x, y))
 
-@implements(np.nextafter, module='numpy')
+
 @partial(jit, inline=True)
 def nextafter(x: ArrayLike, y: ArrayLike, /) -> Array:
+  """Return element-wise next floating point value after ``x1`` towards ``x2``.
```
Comment (jakevdp) [inline jax/_src/numpy/ufuncs.py:924]:
```suggestion
  """Return element-wise next floating point value after ``x`` towards ``y``.
```

File: jax/_src/numpy/ufuncs.py [hunk 2]
```diff
@@ -872,14 +872,76 @@ def maximum(x: ArrayLike, y: ArrayLike, /) -> Array:
...
+
 @partial(jit, inline=True)
 def float_power(x: ArrayLike, y: ArrayLike, /) -> Array:
+  """Calculate element-wise base ``x`` exponential of ``y``.
+
+  JAX implementation of :obj:`numpy.float_power`.
+
+  Args:
+    x: scalar or array. Specifies the bases.
+    y: scalar or array. Specifies the exponents. Both ``x`` and ``y`` should
```
Comment (jakevdp) [inline jax/_src/numpy/ufuncs.py:884]: This reads strangely to me. Maybe remove the word "Both"?
```suggestion
    y: scalar or array. Specifies the exponents. ``x`` and ``y`` should
```

Same comment in the docstring below.

File: jax/_src/numpy/ufuncs.py [hunk 3]
```diff
@@ -872,14 +872,76 @@ def maximum(x: ArrayLike, y: ArrayLike, /) -> Array:
...
+    Array([ 9. ,  1. , -0.2], dtype=float32)
+
+    Inputs with broacast compatibility:
+
+    >>> x1 = jnp.array([[2, -4, 1],
+    ...                 [-1, 2, 3]])
+    >>> y1 = jnp.array([-2, 1, 4])
+    >>> jnp.float_power(x1, y1)
+    Array([[ 0.25, -4.  ,  1.  ],
+          [ 1.  ,  2.  , 81.  ]], dtype=float32)
```
Comment (jakevdp) [inline jax/_src/numpy/ufuncs.py:911]: Output formatting looks off (will likely be caught by doctests)

File: jax/_src/numpy/ufuncs.py [hunk 4]
```diff
@@ -872,14 +872,76 @@ def maximum(x: ArrayLike, y: ArrayLike, /) -> Array:
...
+
+  Returns:
+    An array containing the next representable number of ``x`` in the direction
+    of ``y``.
+
+  Examples:
+    >>> eps = jnp.finfo(jnp.float32).eps
+    >>> x = jnp.array([3, -2, 1])
+    >>> y = jnp.array([2, -1, 2])
+    >>> jnp.allclose(jnp.nextafter(x, y), x+eps)
```
Comment (jakevdp) [inline jax/_src/numpy/ufuncs.py:942]: `allclose` has a tolerance larger than `eps`, so this is not particularly informative.

But that aside, this seems to imply to the reader that `nextafter(x, y)` is expected to be equivalent to `x + eps`, which is not generally the case except when `x = 1.0` and `y` is larger than `x`.

**Review by jakevdp — CHANGES_REQUESTED** [review]

File: jax/_src/numpy/ufuncs.py (hunk 2 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/ufuncs.py]: Done

File: jax/_src/numpy/ufuncs.py (hunk 3 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/ufuncs.py]: Fixed the formatting

File: jax/_src/numpy/ufuncs.py (hunk 4 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/ufuncs.py]: Thanks! Removed the comparison with `x+eps` and just printed the output. Also skipped the doctests.

**Comment by rajasekharporeddy** [comment]
Squashed the commits.

**Review by jakevdp — APPROVED** [review]
Thanks!

**Pull request merged**